File: LoanMe/bank.erl
%%%-------------------------------------------------------------------
%%% @author Divyansh
%%% @copyright (C) 2019, <COMPANY>
%%% @doc
%%%
%%% @end
%%% Created : 18. Jun 2019 20:05
%%%-------------------------------------------------------------------
-module(bank).
-author("Divyansh").

%% API
-export([childBank/3]).

childBank(Bank_name, Balance, Customers) ->
  receive
    {Parent_customer, Amount} ->
      if (Balance-Amount) >= 0 ->
        Parent_customer ! {approve, Amount},
        childBank(Bank_name, Balance - Amount, Customers);
      true ->
        Parent_customer ! {deny},
        childBank(Bank_name, Balance, Customers - 1)
      end

  after 2000 ->
      io:format("~p has ~p dollar(s) remaining. ~n ", [Bank_name, Balance])
  end.

File: LoanMe/bank.java
import java.util.HashMap;
import java.util.Map;

/**
 * @author Divyansh
 *
 */
public class bank extends Thread{
	String bank_name;
	int balance, customers;
	private volatile boolean flag = true;
	
	public bank(String bank_name, int balance, int customers) {
		this.bank_name = bank_name;
		this.balance = balance;
		this.customers = customers;
		start();
	}
	
	public void run() {
		while(flag) {
			
		}
	}
	
	synchronized public String request(int amount) {
		if((balance-amount)>=0) {
			balance = balance - amount;
			return "approve";
		}
		else {
			return "deny";
		}
		
	}
	
	public void exitThread() {
		flag = false;
	}
	
	public void stopThread() {
		customers = customers - 1;
		if(customers==0) {
			System.out.println(bank_name+" has "+balance+" dollar(s) remaining.");
			exitThread();
		}
	}
}
